.travis.yml

```diff
@@ -28,17 +31,27 @@ addons:
       - llvm-toolchain-precise
 
 env:
-  - COMPILER=clang++-3.6 CMAKE_GENERATOR="Ninja"
-  - COMPILER=clang++-3.6 CMAKE_GENERATOR="Unix Makefiles"
-
-  - COMPILER=clang++-3.7 CMAKE_GENERATOR="Unix Makefiles"
-  - COMPILER=clang++-3.7 CMAKE_GENERATOR="Ninja"
-
-  - COMPILER=clang++-3.8 CMAKE_GENERATOR="Unix Makefiles"
-  - COMPILER=clang++-3.8 CMAKE_GENERATOR="Ninja"
-
-  - COMPILER=gcc CMAKE_GENERATOR="Ninja"
-  - COMPILER=gcc CMAKE_GENERATOR="Unix Makefiles"
+  # Test all generators and Clang + GCC
+  - TEST=true COMPILER=clang++-3.6 CMAKE_GENERATOR="Ninja"
+  - TEST=true COMPILER=clang++-3.6 CMAKE_GENERATOR="Unix Makefiles"
+  - TEST=true COMPILER=clang++-3.7 CMAKE_GENERATOR="Ninja"
+  - TEST=true COMPILER=clang++-3.7 CMAKE_GENERATOR="Unix Makefiles"
+  - TEST=true COMPILER=clang++-3.8 CMAKE_GENERATOR="Ninja"
+  - TEST=true COMPILER=clang++-3.8 CMAKE_GENERATOR="Unix Makefiles"
+  - TEST=true COMPILER=g++-4.7     CMAKE_GENERATOR="Ninja"
+  - TEST=true COMPILER=g++-4.7     CMAKE_GENERATOR="Unix Makefiles"
+  - TEST=true COMPILER=g++-4.8     CMAKE_GENERATOR="Ninja"
+  - TEST=true COMPILER=g++-4.8     CMAKE_GENERATOR="Unix Makefiles"
+  - TEST=true COMPILER=g++-4.9     CMAKE_GENERATOR="Ninja"
+  - TEST=true COMPILER=g++-4.9     CMAKE_GENERATOR="Unix Makefiles"
```

Comment:
In all cases, I'm not too worried about bumping the compiler version once in a while, so don't sweat too much over it. I mean, it would be nice if that could be automated, but otherwise it's just a minor annoyance.